pkg/engine/helpers.go

```diff
@@ -35,10 +35,21 @@ func downloadUpdateConfigFile(urlStr string, existsAlready bool) (bool, error) {
 	defer resp.Body.Close()
 	newBytes, err := io.ReadAll(resp.Body)
 	if err != nil {
-		return false, err
+		return false, fmt.Errorf("error downloading config from %s: %v", err)
+	}
+	if newBytes == nil && !initial {
+		return false, fmt.Errorf("found empty config at %s, unable to update config", urlStr)
+	}
+	// if initial, this is the last resort, newBytes should be populated
+	// the only way to get here from initial
+	// is if there is no config file on disk, only a HARPOON_CONFIG_URL
+	if initial {
+		if err := os.WriteFile(defaultConfigPath, newBytes, 0600); err != nil {
+			return false, fmt.Errorf("could not copy contents of %s to path %s: %v", urlStr, defaultConfigPath, err)
+		}
+		return true, nil
```

Comment:
this is the fix, without this, the contents of the config url were never placed in config file, so harpoon failed - in testing b4, I must've had a stale config file at ~/.harpoon/config.yaml so didn't catch this. I've added a workflow for it now.